### PR TITLE
add controller for storage

### PR DESCRIPTION
**Pull request by octavianor**

### CODE REVIEW

File: src/omotes_simulator_core/entities/network_controller.py [hunk 1]
```diff
@@ -105,6 +138,48 @@ def _set_producers_to_max(self) -> dict:
...
+        storages = {}
+        for storage in self.storages:
+            storages[storage.id] = {
+                PROPERTY_HEAT_DEMAND: storage.max_charge_power,
+                PROPERTY_TEMPERATURE_RETURN: storage.temperature_return,
+                PROPERTY_TEMPERATURE_SUPPLY: storage.temperature_supply,
+            }
+        return storages
+
+    def _set_storages_to_demand(self, power: float = 0) -> dict:
```
Comment (samvanderzwan) [inline src/omotes_simulator_core/entities/network_controller.py:169]: This method can either set demand or supply, please rename to for example _set_storages_power

File: unit_test/entities/controller/test_controller_class.py [hunk 2]
```diff
@@ -14,22 +14,19 @@
...
 from omotes_simulator_core.entities.assets.asset_defaults import (
     PROPERTY_HEAT_DEMAND,
     PROPERTY_TEMPERATURE_RETURN,
     PROPERTY_TEMPERATURE_SUPPLY,
     PROPERTY_SET_PRESSURE,
 )
-from omotes_simulator_core.infrastructure.utils import pyesdl_from_file
 
 
 class ControllerTest(unittest.TestCase):
```
Comment (samvanderzwan) [inline unit_test/entities/controller/test_controller_class.py:32]: You now extended this test to also include also storage. However now we lack a test without a storage, which should be included. Also you only add one storage it is better to also test for two storages

File: src/omotes_simulator_core/entities/assets/controller/controller_storage.py [hunk 3]
```diff
@@ -47,23 +48,30 @@ def __init__(
         self.temperature_supply = temperature_supply
         self.profile: pd.DataFrame = profile
         self.start_index = 0
-        self.max_power: float = max_power
+        self.max_charge_power: float = max_charge_power
+        self.max_discharge_power: float = max_discharge_power
 
-    def get_heat_demand(self, time: datetime.datetime) -> float:
-        """Method to get the heat demand of the storage.
+    def get_heat_power(self, time: datetime.datetime) -> float:
```
[the remark this answers is not in the document]
Comment (octavianor) [reply src/omotes_simulator_core/entities/assets/controller/controller_storage.py:54]: yes, if use wants to use user define profiles for charging/discharging

File: src/omotes_simulator_core/entities/network_controller.py [hunk 4]
```diff
@@ -35,10 +35,10 @@ class NetworkController(NetworkControllerAbstract):
     """Class to store the network controller."""
 
     def __init__(
-        self,
-        producers: List[ControllerProducer],
-        consumers: List[ControllerConsumer],
-        storages: List[ControllerStorage],
+            self,
```
[the remark this answers is not in the document]
Comment (octavianor) [reply src/omotes_simulator_core/entities/network_controller.py:38]: oke

File: src/omotes_simulator_core/entities/assets/controller/controller_storage.py [hunk 5]
```diff
@@ -29,13 +29,14 @@ class ControllerStorage(AssetControllerAbstract):
...
 
     def __init__(
-        self,
-        name: str,
-        identifier: str,
-        temperature_supply: float,
-        temperature_return: float,
-        max_power: float,
-        profile: pd.DataFrame,
+            self,
```
[the remark this answers is not in the document]
Comment (octavianor) [reply src/omotes_simulator_core/entities/assets/controller/controller_storage.py:32]: ok

File: src/omotes_simulator_core/entities/network_controller.py [hunk 6]
```diff
@@ -105,6 +138,48 @@ def _set_producers_to_max(self) -> dict:
         producers[self.producers[0].id][PROPERTY_SET_PRESSURE] = True
         return producers
 
+    def _set_storages_discharge_to_max(self) -> dict:
+        """Method to set the storages to the max discharge power.
```
[the remark this answers is not in the document]
Comment (octavianor) [reply src/omotes_simulator_core/entities/network_controller.py:142]: ok

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py [hunk 7]
```diff
@@ -142,12 +142,21 @@ def to_entity(self, esdl_asset: EsdlAssetObject) -> ControllerStorage:
...
+            discharge_power = result[0]
+        if discharge_power == 0:
+            discharge_power = np.inf
+
+        result = esdl_asset.get_property(esdl_property_name="maxChargeRate",
+                                         default_value=np.inf)
+        charge_power = np.inf
+        if result[1]:
+            charge_power = result[0]
+        if charge_power == 0:
```
[the remark this answers is not in the document]
Comment (octavianor) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py:158]: indeed, i think user should be able to put zero then

[the remark this answers is not in the document]
Comment (samvanderzwan) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py]: I made an issue to do this

**Review by samvanderzwan — APPROVED** [review]
With the changes it is ok

**Pull request merged**